packages/transaction-controller/src/TransactionController.ts

```diff
@@ -789,23 +789,34 @@ export class TransactionController extends BaseController<
    *
    * @param ignoreNetwork - Determines whether to wipe all transactions, or just those on the
    * current network. If `true`, all transactions are wiped.
+   * @param address - If specified, only transactions originating from this address will be
+   * wiped on current network.
    */
-  wipeTransactions(ignoreNetwork?: boolean) {
+  wipeTransactions(ignoreNetwork?: boolean, address?: string) {
     /* istanbul ignore next */
-    if (ignoreNetwork) {
+    if (ignoreNetwork && !address) {
       this.update({ transactions: [] });
       return;
     }
     const { providerConfig, networkId: currentNetworkID } =
       this.getNetworkState();
     const { chainId: currentChainId } = providerConfig;
     const newTransactions = this.state.transactions.filter(
-      ({ networkID, chainId }) => {
+      ({ networkID, chainId, transaction }) => {
         // Using fallback to networkID only when there is no chainId present. Should be removed when networkID is completely removed.
-        const isCurrentNetwork =
+        const isMatchingNetwork =
```

Comment:
As we are working around transaction-controller for a long time, maybe in a future PR we could use the `transactionMatchesNetwork` [here](https://github.com/MetaMask/core/blob/f979ae556d95864c6ed384b3fe34b8abedf346d7/packages/transaction-controller/src/utils.ts#L221) once my PR is in.

Reply:
Sure that makes sense, I will do it in one of the upcoming PRs after yours merged 👍